routesrv/eskipbytes.go

```diff
@@ -11,10 +11,35 @@ import (
 
 	ot "github.com/opentracing/opentracing-go"
 	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/metrics"
 	"github.com/zalando/skipper/routing"
 	"github.com/zalando/skipper/tracing"
 )
 
+type responseWriterInterceptor struct {
```

Comment:
This will break all usecases for Flusher, Pusher, CloseNotifier, and Hijacker.

Reply:
I guess it is not important for our usecase. Its a pity that the only way to capture status is to wrap ResponseWriter.

Here are some interesting related links:
* https://stackoverflow.com/questions/53272536/how-do-i-get-response-statuscode-in-golang-middleware
* https://github.com/kubernetes/kubernetes/blob/a9593d634c6a053848413e600dadbf974627515f/staging/src/k8s.io/apiserver/pkg/endpoints/responsewriter/wrapper.go#L37-L96
* https://github.com/golang/go/issues/54136#issuecomment-1259894379

Reply:
Kubernetes code looks like Kubernetes code :)
The SO fixes just half of the problem.